### DISCUSSION TITLE
Celery with SSL enabled unable to connect to redis

### DISCUSSION THREAD

**Discussion by ghost** [apache/airflow#38622] [Q&A]
Hi Everyone,

I have enabled SSL on the airflow webserver and celery worker. Once the SSL is enabled worker is unable to connect to redis host. Below is the config and logs for SSL enabled and disabled for comparison.

airflow config values -
  AIRFLOW__WEBSERVER__WEB_SERVER_SSL_CERT: "/path/to/cert/dataworkz_subdomain_cert.crt"
  AIRFLOW__WEBSERVER__WEB_SERVER_SSL_KEY: "/path/to/cert/dataworkz_subdomain_key.key"
  AIRFLOW__CELERY__SSL_ACTIVE: True
  AIRFLOW__CELERY__SSL_KEY: "/path/to/certdataworkz_subdomain_key.key"
  AIRFLOW__CELERY__SSL_CERT: "/path/to/cert/dataworkz_subdomain_cert.crt"
  AIRFLOW__CELERY__SSL_CACERT: "/path/to/cert/ca_cert.crt"

Worker logs with https disabled -
[2024-03-29 10:55:52 +0000] [72] [INFO] Starting gunicorn 20.1.0
[2024-03-29 10:55:52 +0000] [72] [INFO] Listening at: http://[::]:8793 (72)
[2024-03-29 10:55:52 +0000] [72] [INFO] Using worker: sync
[2024-03-29 10:55:52 +0000] [73] [INFO] Booting worker with pid: 73
[2024-03-29 10:55:52 +0000] [74] [INFO] Booting worker with pid: 74
 
 -------------- celery@dworkz-airflow-worker-0 v5.2.7 (dawn-chorus)
--- ***** ----- 
-- ******* ---- Linux-5.15.0-1058-azure-x86_64-with-glibc2.2.5 2024-03-29 10:55:52
- *** --- * --- 
- ** ---------- [config]
- ** ---------- .> app:         airflow.executors.celery_executor:0x7f54c8580a00
- ** ---------- .> transport:   redis://:**@dworkz-airflow-redis-master.dataworkz.svc.cluster.local:6379/1
- ** ---------- .> results:     postgresql://dwairflowdbadmin:**@172.29.144.4:5432/postgres
- *** --- * --- .> concurrency: 16 (prefork)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** ----- 
 -------------- [queues]
                .> default          exchange=default(direct) key=default
                

[tasks]
  . airflow.executors.celery_executor.execute_command

[2024-03-29 10:55:54,892: INFO/MainProcess] Connected to redis://:**@dworkz-airflow-redis-master.dataworkz.svc.cluster.local:6379/1
[2024-03-29 10:55:54,899: INFO/MainProcess] mingle: searching for neighbors
[2024-03-29 10:55:55,916: INFO/MainProcess] mingle: all alone
[2024-03-29 10:55:55,953: INFO/MainProcess] celery@dworkz-airflow-worker-0 ready.

Worker logs with https enabled-
[2024-03-29 12:06:28 +0000] [71] [INFO] Starting gunicorn 20.1.0
[2024-03-29 12:06:28 +0000] [71] [INFO] Listening at: http://[::]:8793 (71)
[2024-03-29 12:06:28 +0000] [71] [INFO] Using worker: sync
[2024-03-29 12:06:28 +0000] [72] [INFO] Booting worker with pid: 72
[2024-03-29 12:06:28 +0000] [73] [INFO] Booting worker with pid: 73
 
 -------------- celery@dworkz-airflow-worker-0 v5.2.7 (dawn-chorus)
--- ***** ----- 
-- ******* ---- Linux-5.15.0-1058-azure-x86_64-with-glibc2.2.5 2024-03-29 12:06:28
- *** --- * --- 
- ** ---------- [config]
- ** ---------- .> app:         airflow.executors.celery_executor:0x7f7593e8d9d0
- ** ---------- .> transport:   redis://:**@dworkz-airflow-redis-master.dataworkz.svc.cluster.local:6379/1
- ** ---------- .> results:     postgresql://dwairflowdbadmin:**@172.29.144.4:5432/postgres
- *** --- * --- .> concurrency: 16 (prefork)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** ----- 
 -------------- [queues]
                .> default          exchange=default(direct) key=default
                

[tasks]
  . airflow.executors.celery_executor.execute_command

**Comment by potiuk** [comment]
Likely your Redis is not configured for SSL. I have never done that but the configuration for airflow connecting to redis seems like client connection where you speciffy what certificates you use to connect and in order to connect to Redis over SSL you also need to configure redis to a) use SSL b) accept the certificates c) present the certificates that airflow celery client will accept. This is a generic knowledge of how you usually configure MTLS (mutual TLS) connections (so I might be wrong). I guess you should consult and understand Redis SSL connectivity for this to work. 

What does the Redis SSL configuration doc say? Did you found and configure Redis for SSL accepting the connection as well?`
